### PR TITLE
post and get events created

### PR DESCRIPTION
**Pull request by mubeenfayyaz20**

### CODE REVIEW

File: db.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{
```
Comment (ahsan-bilal-786) [inline db.json]: please add this file in gitignore and donot push it in repository.

File: src/components/EventManagement/AddNewEvents.jsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const newDateTime = new Date();
+    setStartTime(newDateTime);
+    const newDateTimeEnd = new Date();
+    setEndTime(newDateTimeEnd);
+    console.log(title, weddingDecription, startTime, endTime);
+    try {
+      const postData = await fetch("http://localhost:3000/events", {
```
Comment (ahsan-bilal-786) [inline src/components/EventManagement/AddNewEvents.jsx]: please create an env file and add the API domain `http://localhost:3000/` in env and use `process.env` attributes to use the domain with API paths.

Comment (ahsan-bilal-786) [reply src/components/EventManagement/AddNewEvents.jsx]: Please create a new file in api directory and keep all the APIs in that file. Each API should be added in a function and use that api function in th component.
```
File Path: <app_root>/src/api/index.js
```

**Review by ahsan-bilal-786 — CHANGES_REQUESTED** [review]

File: src/components/EventManagement/AddNewEvents.jsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  const [startTime, setStartTime] = useState(new Date());
+  const [endTime, setEndTime] = useState(new Date());
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const newDateTime = new Date();
+    setStartTime(newDateTime);
+    const newDateTimeEnd = new Date();
+    setEndTime(newDateTimeEnd);
+    console.log(title, weddingDecription, startTime, endTime);
```
Comment (ahsan-bilal-786) [inline src/components/EventManagement/AddNewEvents.jsx]: never push console logs in your code.

**Review by ahsan-bilal-786 — CHANGES_REQUESTED** [review]

File: db.json (hunk 1 above)
Comment (mubeenfayyaz20) [reply db.json]: done

File: src/components/EventManagement/AddNewEvents.jsx (hunk 3 above)
Comment (mubeenfayyaz20) [reply src/components/EventManagement/AddNewEvents.jsx]: removed

File: src/components/EventManagement/AddNewEvents.jsx (hunk 2 above)
Comment (mubeenfayyaz20) [reply src/components/EventManagement/AddNewEvents.jsx]: done

File: src/components/EventManagement/AddNewEvents.jsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    try {
+      const postData = await eventPostData({
+        title: title,
+        weddingDecription: weddingDecription,
+        startTime: startTime,
+        endTime: endTime,
+      });
+
+      console.log(postData);
+      debugger;
```
Comment (ahsan-bilal-786) [inline src/components/EventManagement/AddNewEvents.jsx]: remove debugger and console from code as these are the test statements and no need to push them to repository.

File: src/components/EventManagement/AllEvents.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from "react";
+import { NavLink } from "react-router-dom";
+import { getEventdata } from "../../api";
+const AllEvents = () => {
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const jsonData = await getEventdata();
+      console.log("json data get", jsonData);
```
Comment (ahsan-bilal-786) [inline src/components/EventManagement/AllEvents.jsx]: remove console log

**Review by ahsan-bilal-786 — CHANGES_REQUESTED** [review]

File: db.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  "events": [
+    {
+      "title": "dasdsa",
+      "weddingDecription": "dsdsa",
```
Comment (ahsan-bilal-786) [inline db.json]: its good, if you added the db in gitignore . Please remove this file from branch as well as the file or data should not be in branch/repo.

**Review by ahsan-bilal-786 — CHANGES_REQUESTED** [review]

File: src/components/EventManagement/AllEvents.jsx (hunk 5 above)
Comment (ahsan-bilal-786) [reply src/components/EventManagement/AllEvents.jsx]: no need to create a new async  function in useEffect. Please  assign useEffect function as async or use the Promise.then() approach to implement the logic.

**Review by ahsan-bilal-786 — CHANGES_REQUESTED** [review]

File: src/components/EventManagement/AddNewEvents.jsx (hunk 4 above)
Comment (mubeenfayyaz20) [reply src/components/EventManagement/AddNewEvents.jsx]: removed

File: src/components/EventManagement/AllEvents.jsx (hunk 5 above)
Comment (mubeenfayyaz20) [reply src/components/EventManagement/AllEvents.jsx]: removed

File: db.json (hunk 6 above)
Comment (mubeenfayyaz20) [reply db.json]: done file removed

File: src/components/EventManagement/AllEvents.jsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+const AllEvents = () => {
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    const getData = new Promise((resolve, reject) => {
+      resolve(getEventdata());
+    }).then((res) => {
+      setData(res);
+    });
+  }, []);
```
Comment (ahsan-bilal-786) [inline src/components/EventManagement/AllEvents.jsx]:
```
useEffect(async () => {
const data =       await getEventdata();
setState(data);
  }, []);
```

```
useEffect(() => {
getEventdata().then((resp) => {
setState(resp);
});

  }, []);
```

File: src/components/EventManagement/AddNewEvents.jsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
+import React, { useState } from "react";
+import { eventPostData } from "../../api";
+
+const AddNewEvents = () => {
+  const [title, setTitle] = useState("");
+  const [weddingDecription, setWeddingDecription] = useState("");
+  const [startTime, setStartTime] = useState(new Date());
+  const [endTime, setEndTime] = useState(new Date());
```
Comment (ahsan-bilal-786) [inline src/components/EventManagement/AddNewEvents.jsx]: use date-fns plugin for date implementation

File: src/components/EventManagement/AllEvents.jsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+            <tr>
+              <th scope="col">#</th>
+              <th scope="col">Title</th>
+              <th scope="col">Description</th>
+              <th scope="col">Time Start</th>
+              <th scope="col">Time End</th>
+            </tr>
+          </thead>
+          <tbody>
+            {data.map((item, index) => {
```
Comment (ahsan-bilal-786) [inline src/components/EventManagement/AllEvents.jsx]: install lodash and use map of lodash

**Review by ahsan-bilal-786 — CHANGES_REQUESTED** [review]

**Review by ahsan-bilal-786 — APPROVED** [review]

File: src/components/EventManagement/AllEvents.jsx (hunk 7 above)
Comment (mubeenfayyaz20) [reply src/components/EventManagement/AllEvents.jsx]: done

File: src/components/EventManagement/AllEvents.jsx (hunk 9 above)
Comment (mubeenfayyaz20) [reply src/components/EventManagement/AllEvents.jsx]: done

File: src/components/EventManagement/AddNewEvents.jsx (hunk 8 above)
Comment (mubeenfayyaz20) [reply src/components/EventManagement/AddNewEvents.jsx]: done

File: src/components/EventManagement/AllEvents.jsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from "react";
+import { NavLink, useNavigate } from "react-router-dom";
+import { getEventdata, eventDelete } from "../../api";
+import _ from "lodash";
```
Comment (ahsan-bilal-786) [inline src/components/EventManagement/AllEvents.jsx]: never import complete lodash to use in a component as it increase the build size. Please import the utility from lodash which is needed,
```
import map from "lodash/map";
```

File: src/components/EventManagement/UpdateEvent.jsx [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+    const fetchSingalData = () => {
+      try {
+        getEventDataID(id).then((res) => {
+          setEventData(res);
+        });
+      } catch (err) {
+        console.log(err);
+      }
+    };
+    fetchSingalData();
```
Comment (ahsan-bilal-786) [inline src/components/EventManagement/UpdateEvent.jsx]: no need to create a new function `fetchSingalData` for fetching the event data. Please keep the logic in useEffect and remove the `fetchSingalData` function.

**Review by ahsan-bilal-786 — CHANGES_REQUESTED** [review]

File: src/components/EventManagement/AllEvents.jsx (hunk 10 above)
Comment (mubeenfayyaz20) [reply src/components/EventManagement/AllEvents.jsx]: done

File: src/components/EventManagement/UpdateEvent.jsx (hunk 11 above)
Comment (mubeenfayyaz20) [reply src/components/EventManagement/UpdateEvent.jsx]: done

**Review by ahsan-bilal-786 — APPROVED** [review]

**Pull request merged**